staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go

```diff
@@ -186,13 +185,13 @@ func NewConflict(qualifiedResource schema.GroupResource, name string, err error)
 }
 
 // NewApplyConflict returns an error including details on the requests apply conflicts
-func NewApplyConflict(conflicts merge.Conflicts) *StatusError {
-	causes := make([]metav1.StatusCause, 0, len(conflicts))
-	for _, conflict := range conflicts {
+func NewApplyConflict(conflicts []string, fields []string, message string) *StatusError {
+	causes := make([]metav1.StatusCause, 0, len(fields))
+	for i, conflict := range conflicts {
 		causes = append(causes, metav1.StatusCause{
-			Type:    metav1.CauseType("conflict"),
-			Message: conflict.Error(),
-			Field:   conflict.Path.String(),
+			Type:    metav1.CauseTypeFieldManagerConflict,
+			Message: conflict,
+			Field:   fields[i],
```

Comment:
That can crash if `conflicts` is longer than `fields`.

Reply:
Hmm, that should never happen though. I can put a comment saying they should always be equal length, or panic maybe?

Reply:
I think: `struct > panic > comment`